### PR TITLE
Extract makeClient functionality from each module and move it into th…

### PR DESCRIPTION
**Pull request by jans-fp**
…e core package.

Rename default client props

Rename files

Fix exports

Fix module names

Improve readme

### CODE REVIEW

File: src/AWS/CloudWatch/Cloudwatch.purs [hunk 1]
```diff
@@ -1,44 +1,38 @@
...
 
-makeClient :: Region -> AccessKeyId -> SecretAccessKey -> Maybe SessionToken -> Effect Cloudwatch
-makeClient r a s t =
-  makeClientImpl
-    { region: (un Region r)
-    , secretAccessKey: (un SecretAccessKey s)
-    , accessKeyId: (un AccessKeyId a)
-    , sessionToken: toNullable $ map (un SessionToken) t
-    }
+makeClient :: forall t4 t5 t6 t7 t8.
```
Comment (realvictorprm) [inline src/AWS/CloudWatch/Cloudwatch.purs:31]: Can we name the type params here beginning from 0? :thinking: 4 feels odd :smile_cat:

File: src/AWS/CloudWatchLogs/CloudwatchLogs.purs [hunk 2]
```diff
@@ -1,46 +1,38 @@
...
 
-makeClient :: Region -> AccessKeyId -> SecretAccessKey -> Maybe SessionToken -> Effect API
-makeClient r a s t =
-  makeClientImpl
-    { region: (un Region r)
-    , secretAccessKey: (un SecretAccessKey s)
-    , accessKeyId: (un AccessKeyId a)
-    , sessionToken: toNullable $ map (un SessionToken) t
-    }
+makeClient :: forall t4 t5 t6 t7 t8.
```
Comment (realvictorprm) [inline src/AWS/CloudWatchLogs/CloudwatchLogs.purs:27]: same here :thinking:

File: src/AWS/CostExplorer/CostExplorer.purs [hunk 3]
```diff
@@ -1,37 +1,34 @@
...
 
-makeClient :: Region -> AccessKeyId -> SecretAccessKey -> Maybe SessionToken -> Effect CE
-makeClient r a s t = do
-  makeClientImpl
-    { region: (un Region r)
-    , secretAccessKey: (un SecretAccessKey s)
-    , accessKeyId: (un AccessKeyId a)
-    , sessionToken: toNullable $ map (un SessionToken) t
-    }
+makeClient :: forall t4 t5 t6 t7 t8.
```
Comment (realvictorprm) [inline src/AWS/CostExplorer/CostExplorer.purs:27]: same here :thinking:

File: src/AWS/EC2/EC2.purs [hunk 4]
```diff
@@ -1,36 +1,30 @@
...
-makeClient r a s t =
-  makeClientImpl
-    { region: (un Region r)
-    , secretAccessKey: (un SecretAccessKey s)
-    , accessKeyId: (un AccessKeyId a)
-    , sessionToken: toNullable $ map (un SessionToken) t
-    }
+foreign import newEC2 :: Foreign -> (Effect EC2)
+
+makeClient :: forall t4 t5 t6 t7 t8.
```
Comment (realvictorprm) [inline src/AWS/EC2/EC2.purs:23]: same here :thinking:

File: src/AWS/Lambda/Lambda.purs [hunk 5]
```diff
@@ -17,31 +22,18 @@ type InternalLambdaParams = {
...
 
-type InternalMakeClientParams
-  = { region :: String
-    , secretAccessKey :: String
-    , accessKeyId :: String
-    }
+foreign import newLambda :: Foreign -> (Effect Lambda)
 
-foreign import makeClientImpl :: InternalMakeClientParams -> (Effect Lambda)
+makeClient :: forall t4 t5 t6 t7 t8.
```
Comment (realvictorprm) [inline src/AWS/Lambda/Lambda.purs:27]: and here :sweat_smile:

File: src/AWS/SecurityTokenService/SecurityTokenService.purs [hunk 6]
```diff
@@ -17,53 +19,48 @@ import Foreign (Foreign)
...
-  Union
-    t3
-    t5
-    PropsR =>
-  RowToList t4 t7 =>
-  JustifiableFields t7 t4 () t3 =>
-  Record t4 ->
-  Effect STS
-makeClient r = ((justifill r) :: Props) # makeClientHelper
+makeClient :: forall t10 t6 t7 t8 t9.
```
Comment (realvictorprm) [inline src/AWS/SecurityTokenService/SecurityTokenService.purs:46]: here wood be good too :thinking: Great mix :smile:

**Review by realvictorprm — APPROVED** [review]
Only this small nit. Rest looks fine to me

File: src/AWS/Lambda/Lambda.purs [hunk 7]
```diff
@@ -17,31 +22,18 @@ type InternalLambdaParams = {
...
-    }
-
-makeDefaultClient :: Effect Lambda
-makeDefaultClient = makeDefaultClientImpl
-
-foreign import invokeFunctionImpl :: forall output. Fn2 Lambda InternalLambdaParams (Effect (Promise output))
-
-invokeFunction :: forall input output. WriteForeign input => Lambda -> Arn -> input -> Aff output
-invokeFunction client (Arn arn) input = runFn2 invokeFunctionImpl client params # toAffE
+invoke :: forall input output. WriteForeign input => Lambda -> Arn -> input -> Aff output
```
Comment (tiphdousset) [inline src/AWS/Lambda/Lambda.purs:35]: Reminder: update the call in the CostExplorer repo (Main.purs of the Importer-proxy)

**Review by tiphdousset — APPROVED** [review]
LGTM. 
I like that our Readme is finally updated :D

**Pull request merged**